test/src/test/java/org/corfudb/runtime/object/transactions/OptimisticTransactionContextTest.java

```diff
@@ -895,4 +909,89 @@ public void txnCanCommitIfOverwrittenAndRetry() throws Exception {
         assertThat(get("k1")).isEqualTo("v1");
         assertThat(get("k2")).isEqualTo("v2");
     }
+    /**
+     *
+     */
+    @Test
+    public void testAffectedStreamsWithStreamTags() {
+        CorfuRuntime rt = getDefaultRuntime().connect();
+
+        // Create a table with a stream tag
+        Map<String, String> testMap1 = rt.getObjectsView()
+            .build().setStreamName("test-1")
+            .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+            .setStreamTags(CorfuRuntime.getStreamID("test-1"))
+            .open();
+
+        // Create another table without custom tag.
+        Map<String, String> testMap2 = rt.getObjectsView()
+            .build().setStreamName("test-2")
+            .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+            .open();
+
+        t1(() -> rt.getObjectsView().TXBegin());
+        t2(() -> rt.getObjectsView().TXBegin());
+        t1(() -> testMap1.put("azusavnj", "1"));
+        t2(() -> testMap2.put("ajkenmbb", "2"));
+
+        t1(() -> {
+            OptimisticTransactionalContext context =
+                (OptimisticTransactionalContext) TransactionalContext
+                    .getCurrentContext();
+            assertThat(context.getModifiedProxies().size() == 1);
+            context.getModifiedProxies().forEach(p ->
+                assertThat(
+            p.getStreamTags().contains(CorfuRuntime.getStreamID("test-1"))));
+        });
+
+        t2(() -> {
+            OptimisticTransactionalContext context =
+                (OptimisticTransactionalContext) TransactionalContext
+                    .getCurrentContext();
+            assertThat(context.getModifiedProxies().isEmpty());
+            assertThat(context.getWriteSetInfo().getAffectedStreams().contains(
+                CorfuRuntime.getStreamID("test-2")));
```

Comment:
assertThat(condition) does not guarantee condition is true. It needs a terminal operation like `assertThat(condition).isTrue()` or `assertThat(condition).isZero()`